src/plugins/share/public/components/url_panel_content.tsx

```diff
@@ -186,13 +203,25 @@ export class UrlPanelContent extends Component<Props, State> {
     return this.props.shareableUrl || window.location.href;
   };
 
-  private makeUrlEmbeddable = (url: string) => {
-    const embedQueryParam = '?embed=true';
+  private getEmbedQueryParams = (): string => {
+    return [
+      ['&show-top-nav-menu=true', this.state.showTopNavMenu],
+      ['&show-query-input=true', this.state.showQueryInput],
+      ['&show-date-picker=true', this.state.showDatePicker],
+      ['&hide-filter-bar=true', !this.state.showFilterBar], // Inverted to keep default behaviour for old links
+    ].reduce(
+      (accumulator, [queryParam, include]) => (include ? accumulator + queryParam : accumulator),
+      '?embed=true'
+    );
+  };
+
+  private makeUrlEmbeddable = (url: string): string => {
     const urlHasQueryString = url.indexOf('?') !== -1;
+    const embedQueryParams = this.getEmbedQueryParams();
     if (urlHasQueryString) {
-      return url.replace('?', `${embedQueryParam}&`);
+      return url.replace('?', `${embedQueryParams}&`);
     }
-    return `${url}${embedQueryParam}`;
+    return `${url}${embedQueryParams}`;
   };
```

Comment:
Once #54403 is merged this would need implementing in src/plugins/share/server/routes/goto.ts  too.

Reply:
This is now solved differently, current solution should be fine.